### PR TITLE
Index suggest data fetch suggestions

### PR DESCRIPTION
**Pull request by felfilali**
Adding Draft to get feedback on refactored indexing classes and Yelp Suggest Test Script.

### CODE REVIEW

File: src/test/java/org/apache/platypus/server/YelpBaseTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.apache.platypus.server.grpc.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Logger;
+
+public class YelpBaseTest {
+    protected static final Logger logger = Logger.getLogger(YelpReviewsTest.class.getName());
```
Comment (umeshdangat) [inline src/test/java/org/apache/platypus/server/YelpBaseTest.java]: rename?

Comment (umeshdangat) [reply src/test/java/org/apache/platypus/server/YelpBaseTest.java]: can me make the name  more representative of what this class does? For example `TestIndexManager`?

File: src/test/java/org/apache/platypus/server/YelpSuggestTest.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                .setMode(Mode.STANDALONE)
+                .setPrimaryGen(0)
+                .setRestore(false)
+                .build();
+        startIndex(standaloneServerClient, startIndexRequest);
+        // index docs
+        long t1 = System.nanoTime();
+        Stream.Builder<AddDocumentRequest> builder = Stream.builder();
+
+        final ExecutorService indexService = YelpReviewsTest.createExecutorService(
```
Comment (umeshdangat) [inline src/test/java/org/apache/platypus/server/YelpSuggestTest.java]: could we move this method to TestIndexManager as well ?

File: src/test/java/org/apache/platypus/server/YelpSuggestTest.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                .build());
+
+        return standaloneServerClient.getBlockingStub().buildSuggest(
+                buildSuggestRequestBuilder.build()
+        );
+    }
+
+    public static void main(String[] args) throws IOException, Exception{
+        // The dir where the indices will live in the remote server
+        Path yelp_suggest_test_base_path = Paths.get("/nail/tmp");
```
Comment (umeshdangat) [inline src/test/java/org/apache/platypus/server/YelpSuggestTest.java]: can we use defaults (e.g. user.home) in lieu of a missing config/args?

File: src/test/java/org/apache/platypus/server/YelpSuggestTest.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+        public String getCountry() {
+            return country;
+        }
+
+        public Integer getReview_count() {
+            return review_count;
+        }
+
+        public Double getReview_wilson_score() {
```
Comment (umeshdangat) [inline src/test/java/org/apache/platypus/server/YelpSuggestTest.java]: fix casing (for other getters to)?

File: src/test/java/org/apache/platypus/server/YelpSuggestTest.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        return standaloneServerClient.getBlockingStub().buildSuggest(
+                buildSuggestRequestBuilder.build()
+        );
+    }
+
+    public static void main(String[] args) throws IOException, Exception{
+        // The dir where the indices will live in the remote server
+        Path yelp_suggest_test_base_path = Paths.get("/nail/tmp");
+        // The client who will be talking to the remote server
+        LuceneServerClient standaloneServerClient = new LuceneServerClient("dev83-uswest1adevc", 55886);
```
Comment (sarthakn7) [inline src/test/java/org/apache/platypus/server/YelpSuggestTest.java]: Can we just use `localhost` here instead ?

Comment (felfilali) [reply src/test/java/org/apache/platypus/server/YelpSuggestTest.java]: I was planning on making this cmd arg before pushing out this test.

**Pull request closed without merging**